### ISSUE TITLE
Is there a way to "PROGRAM foreign_keys=on" be run on every connect?

### ISSUE THREAD

**Issue by wuranbo** [mattn/go-sqlite3#197]
I have to hack the code, or there is already had.

**Comment by wuranbo** [comment]
I saw ConnectHook。but，sqlite3.go:init() called sql.Register.And sql.Register same name driver twice cause a panic.So there is no way to override it with my own driver with a hook.And looks like, the hook is not useable at all?

There must be something wrong to me? How can I do?

**Comment by mattn** [comment]
I'm thinking how may users expect `foreign_keys=off`.

**Comment by wuranbo** [comment]
@mattn I think the answer may be no people..........Looks like sqlite3.h has SQLITE_DBCONFIG_ENABLE_FKEY can force all connection be foreign_keys=on, but I am not sure.

**Comment by domingues** [comment]
I'm doing this:
```go
// register an hook to run on each database connection
sql.Register("sqlite3_with_fk",
	&sqlite3.SQLiteDriver{
		ConnectHook: func(conn *sqlite3.SQLiteConn) error {
			_, err = conn.Exec("PRAGMA foreign_keys = ON", nil)
			return err
		},
	})
db, err = sql.Open("sqlite3_with_fk", "db.sqlite3")
```

**Comment by mattn** [comment]
fixes by #407
